accounting/decimal_test.go

```diff
@@ -1,46 +1,54 @@
 package accounting_test
 
 import (
+	"math/rand"
 	"testing"
 
-	v2accounting "github.com/nspcc-dev/neofs-api-go/v2/accounting"
+	apiaccounting "github.com/nspcc-dev/neofs-api-go/v2/accounting"
 	"github.com/nspcc-dev/neofs-sdk-go/accounting"
 	"github.com/stretchr/testify/require"
 )
 
-func TestDecimalData(t *testing.T) {
-	const v, p = 4, 2
-
+func testDecimalField[T uint32 | int64](
+	t *testing.T,
+	get func(accounting.Decimal) T,
+	set func(*accounting.Decimal, T),
+	getAPI func(*apiaccounting.Decimal) T,
+) {
 	var d accounting.Decimal
-
-	require.Zero(t, d.Value())
-	require.Zero(t, d.Precision())
-
-	d.SetValue(v)
-	d.SetPrecision(p)
-
-	require.EqualValues(t, v, d.Value())
-	require.EqualValues(t, p, d.Precision())
+	require.Zero(t, get(d))
+
+	val := T(rand.Uint64())
+	set(&d, val)
+	require.EqualValues(t, val, get(d))
+	valOther := val + 1
+	set(&d, valOther)
+	require.EqualValues(t, valOther, get(d))
+
+	t.Run("encoding", func(t *testing.T) {
```

Comment:
strange func tree, imo. `TestDecimal_SetValue` -> `testDecimalField` that tests encoding. not critical thought

Reply:
it tests that set value is correctly transmitted over the network

Reply:
okok, just remember that we used to have `t.Run("encoding", ...)` usually that just checks `Marshal`/`Unmarshal` funcs